src/McBopomofo.cpp

```diff
@@ -168,6 +168,27 @@ McBopomofoEngine::McBopomofoEngine(fcitx::Instance* instance)
   keyHandler_ = std::make_unique<KeyHandler>(
       languageModelLoader_->getLM(), languageModelLoader_,
       std::make_unique<KeyHandlerLocalizedString>());
+
+  keyHandler_->setOnAddNewPhrase([this](std::string message) {
+    auto useGitCommit = config_.gitCommitAfterAddingNewPhrase.value();
+    if (!useGitCommit) return;
+    auto location = languageModelLoader_->userDataPath();
+    auto gitPath = config_.gitPath.value().length() > 0
+                       ? config_.gitPath.value()
+                       : kGitPath;
+    std::string command = "";
+    command += gitPath + " init -b master;";
```

Comment:
As we iterate we may open this up to more customization (for example what if I want to use `hg` instead?), but git has already moved on to use `main` as the default branch name. Should we let people customize this if they use a default branch name that is different from git's default?

Reply:
probably we could have git flow here as the default behavior, but also allow `path to scripts` or `hooks/apis` for customization?

Reply:
I modified the implementation. Now the input method runs a shell script instead of calling git directly, and users can change the path or just modify the script.